### PR TITLE
Update docs with valid examples for fetch URIs

### PR DESCRIPTION
**Pull request by meln1k**
Summary: Updated the docs to use the fetch field instead of uris, and provided comprehensive examples.

JIRA issues: MARATHON-8078

### CODE REVIEW

File: docs/docs/application-basics.md [hunk 1]
```diff
@@ -34,28 +34,39 @@ When you define and launch an application, Marathon hands over execution to Meso
 
 ## Using Resources in Applications
 
-To run any non-trivial application, you typically depend on a collection of resources: files and/or archives of files. To manage resource allocation, Marathon has the concept of URIs. URIs use the Mesos fetcher to do the legwork in terms of downloading (and potentially) extracting resources.
+To run any non-trivial application, you typically depend on a collection of resources: files and/or archives of files. 
+To manage resource allocation, Marathon uses the [Mesos fetcher](http://mesos.apache.org/documentation/latest/fetcher/) 
+do the legwork in terms of downloading (and potentially) extracting resources.
```
Comment (ichernetsky) [inline docs/docs/application-basics.md]: nit: `to do the legwork`.

File: docs/docs/application-basics.md [hunk 2]
```diff
@@ -74,21 +89,78 @@ And how this looks in practice shows you the following example: let's assume you
...
+    "fetch": [
+        {
+          "uri": "https://example.com/app.zip",
+          "cache": true
+        }
+    ]
+}
+```
+
+It's also possible to specify path and name of the destination where fetched resource will be stored:
```
Comment (ichernetsky) [inline docs/docs/application-basics.md]: nit: `the fetched resource`

File: docs/docs/application-basics.md [hunk 3]
```diff
@@ -74,21 +89,78 @@ And how this looks in practice shows you the following example: let's assume you
...
+    "fetch": [
+        {
+          "uri": "https://example.com/some-script.sh",
+          "destPath": "cool-script.sh"
+        }
+    ]
+}
+```
+
+Note also that you can specify many resources, not only one. So, for example, you could provide a git repository
```
Comment (ichernetsky) [inline docs/docs/application-basics.md]: nit: `not just one`.

File: docs/docs/application-basics.md [hunk 4]
```diff
@@ -34,28 +34,39 @@ When you define and launch an application, Marathon hands over execution to Meso
...
+          "executable": true
+        }
     ]
 }
 ```
 
-What the example above does: before executing the `cmd`, download the resource `https://example.com/app/cool-script.sh` (via Mesos) and make it available in the application task's sandbox. You can check that these have been downloaded by visiting the Mesos UI and clicking into a Mesos worker node's sandbox where you should now find `cool-script.sh`.
+What the example above does: before executing the `cmd`, download the resource `https://example.com/app/cool-script.sh` 
+(via Mesos) and make it available in the application task's sandbox. You can check that these have been downloaded by 
+visiting the Mesos UI and clicking into a Mesos worker node's sandbox where you should now find `cool-script.sh`.
```
Comment (alenkacz) [inline docs/docs/application-basics.md]: I think the worker is oficially called agent?

File: docs/docs/application-basics.md [hunk 5]
```diff
@@ -34,28 +34,39 @@ When you define and launch an application, Marathon hands over execution to Meso
...
 
-**Note:** As of Mesos v0.22 and above, the fetcher code does not make downloaded files executable by default. In the example above, `cmd` first makes the file executable.
+As of Mesos v0.22 and above, the fetcher code does not make downloaded files executable by default.
+ In the example above, we should provide the field `"executable": true` to tell Mesos change the fetch result to
```
Comment (alenkacz) [inline docs/docs/application-basics.md]: we provided maybe... ?

File: docs/docs/application-basics.md [hunk 6]
```diff
@@ -65,7 +76,11 @@ As already mentioned above, Marathon also [knows how to handle](https://github.c
 * `.tar.xz`
 * `.zip`
 
-And how this looks in practice shows you the following example: let's assume you have an application executable in a zip file at `https://example.com/app.zip`. This zip file contains the script `cool-script.sh` and that's what you want to execute. Here's how:
+To prevent this behaviour, field `"extract": false` must be provided.
+
+And how this looks in practice shows you the following example: let's assume you have an application executable in a zip
+ file at `https://example.com/app.zip`. This zip file contains the script `cool-script.sh` and that's what you 
+ want to execute. Here's how:
```
Comment (alenkacz) [inline docs/docs/application-basics.md]: the newline here is intentional?

File: docs/docs/application-basics.md [hunk 7]
```diff
@@ -74,21 +89,78 @@ And how this looks in practice shows you the following example: let's assume you
...
 }
 ```
 
 Note that in contrast to the example `basic-1` we now have a `cmd` that looks as follows: `app/cool-script.sh`. This stems from the fact that when the zip file gets downloaded and extracted, a directory `app` according to the file name `app.zip` is created where the content of the zip file is extracted into.
 
-Note also that you can specify many resources, not only one. So, for example, you could provide a git repository and some resources from a CDN as follows:
+Additionally, if you want to use a fetcher cache, `"cache": true` field must be specified. If a URI is encountered 
+for the first time (for the same user), it is first downloaded into the cache, then copied to the sandbox directory 
+from there. If the same URI is encountered again, and a corresponding cache file is resident in the cache or 
+still en route into the cache, then downloading is omitted and the fetcher proceeds directly to copying from the cache.
```
Comment (zen-dog) [inline docs/docs/application-basics.md]: I think it's worth mentioning that the cache is "per agent" and so effectively if the task is restarted on the new agent each time, the file will be downloaded each time.

**Pull request merged**